### PR TITLE
feat: export vercel component into standalone 

### PR DESCRIPTION
**Pull request by baseplate-admin**

### CODE REVIEW

File: src/lib/components/shared/vercel_hover.svelte [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        GLIDER_TRANSITION_DURATION = 200,
+        is_hovered = false,
+        mouse_leave_timeout: NodeJS.Timer | null = null;
+
+    const handle_mouseenter = (event: MouseEvent) => {
+        const target = event.target as HTMLElement;
+        const target_computed_style = getComputedStyle(target);
+
+        // To make sure our operations are proper we need to make sure that the `position` is set to relative
+        glider_container_element.style.position = "relative";
```
Comment (tokitou-san) [inline src/lib/components/shared/vercel_hover.svelte:18]: Why not use `relative` class?

Comment (baseplate-admin) [reply src/lib/components/shared/vercel_hover.svelte:18]: Ease of use and there can be relative passed down from the parent

File: src/routes/+layout.svelte [hunk 2]
```diff
@@ -385,12 +357,13 @@
...
 
-                    <div class="relative mt-[2.8125vw] flex flex-col items-center gap-[0.75vw]">
-                        <active_glider
-                            bind:this={hover_glider_element}
-                            class="absolute h-[4vw] w-[4vw] rounded-[0.75vw] bg-white/10 opacity-0 ease-in-out duration-{GLIDER_TRANSITION_DURATION}"
-                        />
-
+                    <VercelHover
+                        direction="vertical"
+                        glider_container_class="mt-[2.8125vw] flex flex-col items-center gap-[0.75vw]"
```
Comment (tokitou-san) [inline src/routes/+layout.svelte:362]: add `relative` class here?

File: src/lib/components/shared/vercel_hover.svelte [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+        // Do some magic here to get the target's height and width
+        // Don't change the position of this code.
+        // It will cause animation jank
+        hover_glider_element.style.height = target_computed_style.height;
+        hover_glider_element.style.width = target_computed_style.width;
+
+        // We need to make sure that zIndex is not auto
+        const target_zindex = parseInt(target_computed_style.zIndex);
+        hover_glider_element.style.zIndex = String(target_zindex ? target_zindex - 1 : -1);
```
Comment (tokitou-san) [inline src/lib/components/shared/vercel_hover.svelte:28]: Also why these codes?
add `-z-10` class? we're trying to reduce js codes, right?

Comment (baseplate-admin) [reply src/lib/components/shared/vercel_hover.svelte:28]: Bad idea

Comment (baseplate-admin) [reply src/lib/components/shared/vercel_hover.svelte:28]: Bad idea

File: src/routes/+layout.svelte (hunk 2 above)
Comment (baseplate-admin) [reply src/routes/+layout.svelte:362]: Thats what i dont wanna deal with. There will be a nest of problem down the line.

**Pull request merged**